hedera-mirror-rest/stateproof.js

```diff
@@ -0,0 +1,265 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const config = require('./config');
+const constants = require('./constants');
+const EntityId = require('./entityId');
+const TransactionId = require('./transactionId');
+const s3client = require('./s3client');
+const {DbError} = require('./errors/dbError');
+const {NotFoundError} = require('./errors/notFoundError');
+const {FileDownloadError} = require('./errors/fileDownloadError');
+
+/**
+ * Get the consensus_ns of the transaction. Throws exception if no such successful transaction found or multiple such
+ * transactions found.
+ * @param {TransactionId} transactionId
+ * @returns {Promise<String>} consensus_ns of the successful transaction if found
+ */
+let getSuccessfulTransactionConsensusNs = async (transactionId) => {
+  const sqlParams = [transactionId.getEntityId().getEncodedId(), transactionId.getValidStartNs()];
+  const sqlQuery = `SELECT consensus_ns
+       FROM transaction
+       WHERE payer_account_id = $1
+         AND valid_start_ns = $2
+         AND result = 22`; // only the successful transaction
+  if (logger.isTraceEnabled()) {
+    logger.trace(`getSuccessfulTransactionConsensusNs: ${sqlQuery}, ${JSON.stringify(sqlParams)}`);
+  }
+
+  let result;
+  try {
+    result = await pool.query(sqlQuery, sqlParams);
+  } catch (err) {
+    throw new DbError(err.message);
+  }
+
+  const {rows} = result;
+  if (_.isEmpty(rows)) {
+    throw new NotFoundError('Transaction not found');
+  } else if (rows.length > 1) {
+    throw new DbError('Invalid state, more than one transaction found');
+  }
+
+  return _.first(rows).consensus_ns;
+};
+
+/**
+ * Get the RCD file name where consensusNs is in the range [consensus_start, consensus_end]. Throws exception if no
+ * such RCD file found.
+ * @param {string} consensusNs consensus timestamp within the range of the RCD file to search
+ * @returns {Promise<String>} RCD file name
+ */
+let getRCDFileNameByConsensusNs = async (consensusNs) => {
+  const sqlParams = [consensusNs];
+  const sqlQuery = `SELECT name
+       FROM record_file
+       WHERE consensus_end >= $1
+       LIMIT 1`;
+  if (logger.isTraceEnabled()) {
+    logger.trace(`getRCDFileNameByConsensusNs: ${sqlQuery}, ${JSON.stringify(sqlParams)}`);
+  }
+
+  let result;
+  try {
+    result = await pool.query(sqlQuery, sqlParams);
+  } catch (err) {
+    throw new DbError(err.message);
+  }
+
+  const {rows} = result;
+  if (_.isEmpty(rows)) {
+    throw new NotFoundError('No matching RCD file found');
```

Comment:
nit: might be good to say for which consensusNs and therefore which transactionId. 
```suggestion
    throw new NotFoundError(`No matching RCD file found with ${consensusNs} is in the range`);
```

Reply:
updated as suggested